src/command.rs

```diff
@@ -23,8 +23,8 @@ fn test_run_command() {
 #[test]
 fn test_run_command_fail() {
   let result = run_command("asldfkjh test");
-  assert_eq!(
+  assert!(matches!(
     result,
-    Err(String::from("bash: asldfkjh: command not found\n"))
-  );
+    Err(e) if e.contains("command not found") && e.contains("asldfkjh"),
```

Comment:
what's the effect of this post-if syntax? Like what happens if the if conditions return false?

Reply:
if `if` returns false, `matches` returns false
this is a macro for this
```rust
match result {
    Err(e) if e.contains("command not found") && e.contains("asldfkjh") => true,
    _ => false,
}
```

Reply:
makes sense!